nibabel/streamlines/header.py

```diff
@@ -0,0 +1,19 @@
+
+class Field:
```

Comment:
A common use case: Suppose a user created some streamlines and want to save them in a TRK file. In addition to the Tractogram object, he would need to provide header details like the affine, dimension and voxel size. In TRK, we know the field name is `vox_to_ras` but it is/might be different for another format.

Also, the header would be a way of providing automatic "header translation" between different formats. Having a common header is a good way to accomplish that. Right now, I'm using a `dict` with "common" keys contained in `Field`. I'm not against a `TractogramHeader` class (check the one I had in the [previous PR](https://github.com/nipy/nibabel/pull/243/files#diff-3088f83d47b55090ba45b250ca5639af))

So, to  answer your questions.

> You wanted to avoid typos in the nibabel code?
Yes it's basically to avoid typos in nibabel code and make it easier to support other file formats. That said, I realize that adding new file format to the framework won't happen often.

> Or for other people using the code?
People creating tractogram might need to use it. We should probably make some util functions that would it for them though. They would only need to provide a tractogram object, a file name and a reference (e.g. an anat). 

> Do you want to restrict the fields to just these fields? In that case you could just have a class with __slots__, or a numpy structured array, which would do the type checking and field name checking automatically.
No, we should allow conserving additional information in the header as not every file format provides the same information.